config/config.go

```diff
@@ -213,6 +214,7 @@ type (
 	Chain struct {
 		ChainDBPath string `yaml:"chainDBPath"`
 		TrieDBPath  string `yaml:"trieDBPath"`
+		UseBadgerDB bool   `yaml:"useBadgerDB"`
```

Comment:
Can we avoid this flag. Instead, we should use constructor option with Badger (also Bolt and InMem)

Reply:
do you mean moving this to DB config? In the future, we may use different db configs for chain and trie. We may update our current config to
`
    Chain struct {

        ChainDB DB

        StateDB DB

...
`
for now, if we want to easily switch between different db, we may need this flag.

Reply:
will open another PR to address this